clang/test/SemaSYCL/unique_stable_name.cpp

```diff
@@ -0,0 +1,216 @@
+// RUN: %clang_cc1 %s --std=c++17 -triple x86_64-pc-windows-msvc -fsycl-is-device -verify -fsyntax-only -Wno-unused
+// RUN: %clang_cc1 %s --std=c++17 -triple x86_64-linux-gnu -fsycl-is-device -verify -fsyntax-only -Wno-unused
+
+template <typename KernelName, typename KernelType>
+[[clang::sycl_kernel]] void kernel_single_task(KernelType kernelFunc) { // #kernelSingleTask
+  kernelFunc();
+}
+
+// kernel3 - expect error
+// The current function is named with a lambda (i.e., takes a lambda as a
+// template parameter. Call the builtin on the current function
+// Then it is passed to kernel
+// Test that passing the current function to the unique stable name builtin and
+// then to the kernel throws an error because the latter causes its name mangling
+// to change
```

Comment:
```suggestion
// The current function is named with a lambda (i.e., takes a lambda as a
// template parameter). Call the builtin on the current function, then it is
// passed to a kernel. Test that passing the given function to the unique
// stable name builtin and then to the kernel throws an error because the
// latter causes its name mangling to change.
```
Grammar, punctuation, and reflowing the comments.

For the other comments in this batch, I'll fix up the grammar and punctuation, but can you reflow the comments to the usual 80-col limits when you make the suggested changes?